standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1122,13 +1364,30 @@ private ConnectionLockIdPair enqueueLockWithRetry(LockRequest rqst) throws NoSuc
             String dbName = lc.getDbname();
             String tblName = lc.getTablename();
             String partName = lc.getPartitionname();
+            Long writeId = null;
+            if (tblName != null) {
+              // It is assumed the caller have already allocated write id for adding/updating data to
+              // the acid tables. However, DDL operatons won't allocate write id and hence this query
+              // may return empty result sets.
+              // Get the write id allocated by this txn for the given table writes
+              s = "select t2w_writeid from TXN_TO_WRITE_ID where"
```

Comment:
Not all callers able to pass it. For instance writeId allocated by CTAS/dynamic partition won't be available when we build LockRequest.

Reply:
I don't understand this.  Why would DP insert not have a write id?  If acquires locks (which it does) it should have a writeid....

Reply:
What I meant is, DP/CTAS will allocate write ID and is stored in LoadTableDesc which is not available when we build LockRequest. So, it will be complex to get write Id there and pass it here.

Reply:
ok